qa/workunits/rbd/rbd_mirror_helpers.sh

```diff
@@ -907,7 +931,8 @@ create_image_and_enable_mirror()
     fi
 
     create_image ${cluster} ${pool} ${image} $@
-    if [ "${MIRROR_POOL_MODE}" = "image" ] || [ "$pool" = "${PARENT_POOL}" ]; then
+    if [ "${MIRROR_POOL_MODE}" = "image" ] || [ "${pool}" = "${PARENT_POOL}" ] ||
+       [[ "${pool}" = *"${NS2}" ]]; then
```

Comment:
It looks like my comments about this part have been lost after repush.

My concern was about using ` "${pool}" = *"${NS2}"`, if it is bash specific or not. If it is bash specific, then we should use bash in the shebang (not /bin/sh), otherwise it may fail on a sys that does not have /bin/sh as a link to bash. Or use `case` construction instead.

Another suggestion is to create a helper function `is_pool_mode` with this check and use it here like:
```
  if `is_pool_mode ${pool}`; then
    ...
```

Reply:
@trociny 
I tested this approach with "/bin/bash" and "/bin/sh" both works. So you still need this helper function?

Another way of achieving:
```
    if [ "${MIRROR_POOL_MODE}" = "image" ] || [ "${pool}" = "${PARENT_POOL}" ] || $(grep -q ${NS2} <<< "${pool}"); then
    ...
```

I like the existing because it directly checks for suffix without additional keeping.

Reply:
> the helper function is not related to the "bashizm" issue. It is just good to have it and use in two places we already have. It will be more readable and clear what we are checking here for in my opinion.

Sure, I will introduce a helper for this :-)


> When you tested /bin/sh wasn't it a link to bash (as it is on many systems)?

No I mean to say I tested the approach of using '[[' operator approach with below sample code tweaking `bash` and `sh` part
```
#!/bin/bash
string1="hello world"
suffix="world"

if [[ "$string1" = *"${suffix}" ]]; then
  echo "String ends with given suffix."
else
  echo "String does not end with given suffix."
fi
```

Note: I will test the test scripts with journal and snapshot mode soon, the build is baking.

Reply:
> When you tested /bin/sh wasn't it a link to bash (as it is on many systems)?

sorry, You are right!

[0] pkalever 😎 ceph✨ ls /bin/bash -l 
-rwxr-xr-x. 1 root root 1444200 Feb  6  2023 /bin/bash
[0] pkalever 😎 ceph✨ ls /bin/sh -l 
lrwxrwxrwx. 1 root root 4 Feb  6  2023 /bin/sh -> bash

Reply:
@trociny please take a look at the latest version. Thanks!